### ISSUE TITLE
SPIncomingEmailSettings: Incoming email settings

### ISSUE THREAD

**Issue by HeKr09** [PowerShell/SharePointDsc#896]
### Description
The incoming mail settings are not implemented right now. We had to within a project.
### Proposed properties
Ensure (Supposed to use as Enabled-value) [true|false]
UseAutomaticSettings [true|false]
UseDirectoryManagementService [true|false]
RemoteDirectoryManagementService [true|false]
ServerAddress
DLsRequireAuthenticatedSenders [true|false]
DistributionGroupsEnabled [true|false]
ServerDisplayAddress
DropFolder
### Special considerations or limitations
N/A
### Code example to implement
```powershell
SetScript            = {
    Invoke-SPDSCCommand -Arguments @($using:IncomingMailConfiguration, $using:Node.NodeName) -ScriptBlock {
        $config = $args[0]
        $serverName = $args[1]

        try
        {
            $serviceInstance = Get-SPServiceInstance -Server $serverName | Where-Object { $_.TypeName -eq "Microsoft SharePoint Foundation Incoming E-Mail" }
            $InMailService = $serviceInstance.Service

            $InMailService.Enabled = $true
            $InMailService.UseAutomaticSettings = $config.UseAutomaticSettings
            $InMailService.UseDirectoryManagementService = $config.UseDirectoryManagementService
            $InMailService.RemoteDirectoryManagementService = $config.RemoteDirectoryManagementService
            $InMailService.ServerAddress = $config.ServerAddress
            $InMailService.DLsRequireAuthenticatedSenders = $config.DLsRequireAuthenticatedSenders
            $InMailService.DistributionGroupsEnabled = $config.DistributionGroupsEnabled
            $InMailService.ServerDisplayAddress = $config.ServerDisplayAddress
            $InMailService.DropFolder = $config.DropFolder

            $InMailService.Update()
        }
        catch
        {
            Write-Verbose "Could not acivate Incoming Mail Service $($config.DropFolder). $_"
            return $false
        }
    }
}
```

**Comment by ykuijs** [comment]
All yours @MikeLacher448

**Comment by MikeLacher448** [comment]
should have something ready for review next week

**Comment by ykuijs** [comment]
HI @MikeLacher448 Any update?

**Comment by MikeLacher448** [comment]
in the UI, when using the Directory Service integration mode (UseDirectoryManagementService = $true), you set an OU that the farm account has access to create email objects... does anyone know the PowerShell equivalent? I'm not seeing it exposed as a property of the service instance.

**Comment by andikrueger** [comment]
`UseDirectoryManagementService` is a public property of the `SPIncomingEmailService` class. This means you should be able to access this property by running the following code:

```
$service = Get-SPService |  Where-Object -Filterscript {  
        $_.GetType().ToString() -eq "Microsoft.SharePoint.Administration.SPIncomingEmailService"
    }
$service.UseDirectoryManagementService
```

For more details about the service object see: https://docs.microsoft.com/en-us/dotnet/api/microsoft.sharepoint.administration.spincomingemailservice?view=sharepoint-server

**Comment by MikeLacher448** [comment]
yep, that's exactly where I'm looking. I can set that property to True to use the directory management service, but I see no property or method exposed to set the OU in AD to use for creating mail contacts and groups

**Comment by ykuijs** [comment]
Closed via PR #1013
